python/datasources/decia_2020_territory_population.py

```diff
@@ -0,0 +1,336 @@
+}
+
+# used to differentiate renamed columns before melting into HET style df
+TMP_COUNT_SUFFIX: Final = '_count'
+COUNT_CHAR: Final = "C"
+TMP_PCT_SHARE_SUFFIX: Final = '_pct_share'
+PCT_CHAR: Final = "P"
+
+
+class Decia2020TerritoryPopulationData(DataSource):
+    @ staticmethod
+    def get_id():
+        return 'DECIA_2020_TERRITORY_POPULATION_DATA'
+
+    @ staticmethod
+    def get_table_name():
+        return 'decia_2020_territory_population_data'
+
+    def upload_to_gcs(self, gcs_bucket, **attrs):
+        raise NotImplementedError(
+            'upload_to_gcs should not be called for Decia2020TerritoryPopulationData')
+
+    def write_to_bq(self, dataset, gcs_bucket, **attrs):
+
+        # get GEO and DEMO from DAG payload
+        breakdown = self.get_attr(attrs, 'demographic')
+        geo_level = self.get_attr(attrs, 'geographic')
+        raw_dfs_map = load_source_dfs()
+        df = self.generate_breakdown_df(
+            raw_dfs_map, breakdown, geo_level)
+        float_cols = [std_col.POPULATION_COL, std_col.POPULATION_PCT_COL]
+        column_types = gcs_to_bq_util.get_bq_column_types(
+            df, float_cols=float_cols)
+        table_name = f'by_{breakdown}_territory_{geo_level}_level'
+        gcs_to_bq_util.add_df_to_bq(df, dataset, table_name,
+                                    column_types=column_types)
+
+    def generate_breakdown_df(self,
+                              raw_dfs_map: Dict[str, pd.DataFrame],
+                              breakdown: Literal["age", "sex", "race_and_ethnicity"],
+                              geo_level: Literal["state", "county"]):
+        """generate_breakdown_df generates a territory population data frame for a given combo
+         of demographic breakdown and geographic level
+        breakdown: string for type of demographic disaggregation
+        geo_level: string for geographic level (state = territory, county = territory county equivalent) """
+
+        if geo_level == "county":
+            geo_col = std_col.COUNTY_FIPS_COL
+        if geo_level == "state":
+            geo_col = std_col.STATE_FIPS_COL
+
+        rename_map: Dict[str, str] = {}
+        value_cols = []
+        cleaned_dfs: List[pd.DataFrame] = []
+
+        for postal, raw_df in raw_dfs_map.items():
+            raw_df = format_fips_col(raw_df, geo_col)
+
+            # determine relevant values columns and their mapping to HET groups via tmp columns
+            if breakdown == "age":
+                value_cols = get_value_cols(AGE_CODES_TO_STD)
+                rename_map = get_rename_map(AGE_CODES_TO_STD)
+            if breakdown == "sex":
+                value_cols = get_value_cols(SEX_CODES_TO_STD)
+                rename_map = get_rename_map(SEX_CODES_TO_STD)
+            if breakdown == "race_and_ethnicity":
+                value_cols = get_value_cols(RACE_CODES_TO_STD[postal])
+                rename_map = get_rename_map(RACE_CODES_TO_STD[postal])
```

Comment:
Could we use constants for the breakdowns instead of the string?

Reply:
Yes, there are constants in `ingestion/constants.py`